packages/neon-core/src/tx/transaction/PublishTransaction.ts

```diff
@@ -0,0 +1,195 @@
+import { TX_VERSION } from "../../consts";
+import {
+  ab2hexstring,
+  hexstring2str,
+  num2hexstring,
+  num2VarInt,
+  str2hexstring,
+  StringStream
+} from "../../u";
+import { BaseTransaction, TransactionLike } from "./BaseTransaction";
+import TransactionType from "./TransactionType";
+
+export enum ContractParameterType {
+  Signature = 0x00,
+  Boolean = 0x01,
+  Integer = 0x02,
+  Hash160 = 0x03,
+  Hash256 = 0x04,
+  ByteArray = 0x05,
+  PublicKey = 0x06,
+  String = 0x07,
+  Array = 0x10,
+  Map = 0x12,
+  InteropInterface = 0xf0,
+  Void = 0xff
+}
+
+export interface PublishTransactionLike extends TransactionLike {
+  script: string;
+  parameterList: ContractParameterType[];
+  returnType: ContractParameterType;
+  needStorage: boolean;
+  name: string;
+  codeVersion: string;
+  author: string;
+  email: string;
+  description: string;
+}
+
+export interface PublishExclusive {
+  script: string;
+  parameterList: ContractParameterType[];
+  returnType: ContractParameterType;
+  needStorage: boolean;
+  name: string;
+  codeVersion: string;
+  author: string;
+  email: string;
+  description: string;
+}
+
+export class PublishTransaction extends BaseTransaction {
+  public static version: number = 0;
```

Comment:
if this is static, it affects the entire class which doesnt make sense?

Reply:
I can remove the `static` word but if I do, I will get a lot of warnings like this one:

> Declaration of public static method not allowed after declaration of public instance field. Instead, this should come at the beginning of the class/interface. (member-ordering)

Actually initally I just wanted to use the `version` from the `BaseTransaction` via the `this` keyword. But this is not possible because the `version` in the `BaseTransaction` is declared as `public` and
I can't use it in the `static` method `deserializeExclusive` of the subclass `PublishTransaction`.

Reply:
The way that `deserializeExclusive` is used assumes 2 things:
1. The StringStream passed in has the pointer pointing at byte that represents the start of the exclusive data
2. The partial transaction passed in has the previous data deserialized and filled in.

We should be checking the `version` field of the transaction passed in for the version.